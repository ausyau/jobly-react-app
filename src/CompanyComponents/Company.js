<<<<<<< Updated upstream
/* Company component renders multiple jobs related to a company */

import React, { Component } from 'react'
=======
import React, { Component } from 'react';
>>>>>>> Stashed changes
import JoblyApi from '../JoblyApi';
import Jobcard from '../JobComponents/Jobcard';

export default class Company extends Component {
  constructor(props) {
    super(props);

    this.state = {
      name: '',
      description: '',
      jobs: []
    };
  }

  async componentDidMount() {
<<<<<<< Updated upstream
    //using company handle from URL, make an API call and set state from call
    let result = await JoblyApi.getCompany(this.props.match.params.company)
=======
    let result = await JoblyApi.getCompany(this.props.match.params.company);
>>>>>>> Stashed changes
    this.setState({
      name: result.name,
      description: result.description,
      jobs: result.jobs
    });
  }

  componentWillUnmount() {
    this.setState({
      name: '',
      descriptipn: '',
      jobs: []
    });
  }

  render() {
    return (
      <div>
        <h1>{this.state.name}</h1>
        <h3>{this.state.description}</h3>
        {this.state.jobs.map(job => <Jobcard key={job.id} {...job} applyToJob={this.props.applyToJob} user={this.props.user} />)}
      </div>
    );
  }
}
