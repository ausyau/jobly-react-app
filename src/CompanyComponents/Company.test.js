import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";
import Company from "./Company";

it("renders without crashing", function () {
  shallow(<Company />);
});

it("company matches snapshot while loading", function () {
  let wrapper = shallow(<Company />);
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});

it("company matches snapshot after loading", function () {
  let wrapper = shallow(<Company />);
  wrapper.setState( { name: "Specialty's", description: "zomg cookies", jobs: [] });
  let serialized = toJson(wrapper);
  expect(serialized).toMatchSnapshot();
});