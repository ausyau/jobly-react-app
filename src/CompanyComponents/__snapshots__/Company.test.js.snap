// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`company matches snapshot after loading 1`] = `
<div>
  <h1>
    Specialty's
  </h1>
  <h3>
    zomg cookies
  </h3>
</div>
`;

exports[`company matches snapshot while loading 1`] = `
<div>
  <h1 />
  <h3 />
</div>
`;
